package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +137,165 @@ def __init__(self, filename, **kwargs):
+
+        Parameters
+        -----------
+        obj : AtomGroup or Universe
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:
+            raise ImportError("RDKit is required for the RDKitConverter but "
+                              "it's not installed. Try installing it with \n"
+                              "conda install -c conda-forge rdkit")
+        try:
+            # make sure to use atoms (Issue 46)
+            ag = obj.atoms
+        except AttributeError:
+            raise TypeError("No `atoms` attribute in object of type {}, "
+                            "please use a valid AtomGroup or Universe".format(
+                                type(obj))) from None
+
+        try:
+            elements = ag.elements
+        except NoDataError:
+            raise AttributeError(
+                "The `elements` attribute is required for the RDKitConverter "
+                "but is not present in this AtomGroup. Please refer to the "
+                "documentation to guess elements from other attributes or "
+                "type `help(mda.topology.guessers)`") from None
+
+        # attributes accepted in PDBResidueInfo object
+        pdb_attrs = {}
+        for attr in RDATTRIBUTES.keys():
+            if hasattr(ag, attr):
+                pdb_attrs[attr] = getattr(ag, attr)
+        # others
+        other_attrs = {}
+        for attr in ["bfactors", "charges", "segids", "types"]:
+            if hasattr(ag, attr):
+                other_attrs[attr] = getattr(ag, attr)
+
+        mol = Chem.RWMol()
+        atom_mapper = {}
+
+        for i, (atom, element) in enumerate(zip(ag, elements)):
+            # create atom
+            rdatom = Chem.Atom(element.capitalize())
+            # disable adding H to the molecule
+            rdatom.SetNoImplicit(True)
+            # add PDB-like properties
+            mi = Chem.AtomPDBResidueInfo()
+            for attr, values in pdb_attrs.items():
+                _add_mda_attr_to_rdkit(attr, values[i], mi)
+            rdatom.SetMonomerInfo(mi)
+            # other properties
+            for attr in other_attrs.keys():
+                value = other_attrs[attr][i]
+                attr = _TOPOLOGY_ATTRS[attr].singular
+                if isinstance(value, np.float):
+                    rdatom.SetDoubleProp("_MDAnalysis_%s" % attr, float(value))
+                elif isinstance(value, np.int):
+                    rdatom.SetIntProp("_MDAnalysis_%s" % attr, int(value))
+                else:
+                    rdatom.SetProp("_MDAnalysis_%s" % attr, value)
+            rdatom.SetIntProp("_MDAnalysis_index", int(atom.ix))
+            # add atom
+            index = mol.AddAtom(rdatom)
+            # map index in universe to index in mol
+            atom_mapper[atom.ix] = index
+
+        try:
+            bonds = ag.bonds
+            if (len(bonds) == 0) and (ag.n_atoms > 1):
```

Comment:
I think you could simplify this with a call to `ag.bonds.values()` that should raise an IndexError if you have no bonds in your system.

Reply:
I ended up not using bonds.values() as it needs coordinates to work, but people might just want to convert their topology to rdkit without needing the trajectory.

Reply:
That's reasonable.

I'm trying to work out if this does the intended behaviour though. My understanding is that calling `AtomGroup.bonds` should raise a NoDataError if no bond information is present. If bond information is present but the number of bonds is 0, that _should_ be a valid input, at least according to whatever the user provided right?

Is there a particular case where the bonds attribute get populated but we still require guessing?

Reply:
The only case I could find where the bonds attribute was present but empty is a PDB file without a CONECT record, I don't know if that's intended or not (the PDB_helix test file for example).
The corresponding AtomGroup probably still requires guessing bonds though, as I don't really see a scenario where people need to convert their simulation of monoatomic ions or noble gases to RDKit.
For now I can just try if ag.bonds is present if you prefer, PDB files are a mess anyway !

Reply:
That's probably not normal, we should have a consistent API, if bonds aren't present then the bonds attribute doesn't get assigned :/ Can you raise this as an issue too? It might be that there's a rationale for it, but I can't seem to understand what it could be.

Reply:
raised #2832